test/externalTests/README.md

```diff
@@ -0,0 +1,66 @@
+    commits.
+3. Create two new versions of the main branch, representing versions of the compiler currently in
+    `develop` and `breaking`. E.g. if the latest Solidity version is 0.7.5 and the main branch of the
+    external project is called `master`, create `master_070` and `master_080`. This is where we will
+    be adding our own commits. The one corresponding to the newer Solidity version should always be
+    on top on the older one. E.g. if a change is needed to keep the project
+    compilable using Solidity 0.7.x, add it in `master_070` and rebase `master_080` on top of it.
+    If it is only needed for the compiler from its `breaking` branch, add it only in `master_080`.
+    - The fewer commits in these branches, the better. Ideally, any changes needed to make the compiler
+        work should be submitted upstream and the branches should just be tracking the upstream
+        one without any extra commits.
+4. Create a script for compiling/testing the project and put it in `test/externalTests/` in the
+    Solidity repository.
+    - The script should apply workarounds necessary to make the project actually use the compiler
+      binary it receives as a parameter and possibly apply some generic workarounds that should
+      work across different versions of the upstream project. Very specific workarounds that may
+      easily break with every upstream change are better done as commits in the fork.
+5. Add the script to `tests/externalTests.sh`.
+6. Add the test to CircleCI configuration. Make sure to add both a compilation-only run and one that
+    also executes the test suite. If the latter takes a significant amount of time (say, more
+    than 15 minutes) make it run nightly rather than on every PR.
+
+#### Pulling upstream changes into a fork
+1. Pull changes directly into the main branch in the fork. This should be straightforward thanks to
+    it not containing any of our customizations.
+2. If the update is straightforward and looks safe, go straight to point 5. Otherwise you need to
+    test it first.
+3. Create new branches corresponding to the two versioned ones but suffixed with `_new`. E.g.
+    `master_070_new` and `master_080_new`. Then rebase them so that they are on top of the updated
+    main branch. This may require tweaking some of the commits to apply our fixes in new places.
+4. Create PRs on `deveop` and `breaking` in Solidity repo which modify the script to use the new
+    branches instead of `master_070`/`master_080`. Tweak the new branches until external tests
+    in CI pass in the PRs.
+5. Discard the PRs and move the original branches to the place where the `_new` ones are and remove
+    the `_new` branches.
+
+#### Changes needed after a breaking release of the compiler
+When a breaking version of the compiler gets released and becomes the most recent release, the scripts
+and the branches in the forks need to be updated:
```

Comment:
Do we really need to update it? If we merge breaking to develop, the scripts `colony.sh` etc should be the version from breaking.

Reply:
True. That's a great point! Given what @ekpyron also said about [not touching it unless something actually breaks](https://github.com/ethereum/solidity/pull/10556#discussion_r540181402), maybe it's better to have a system where an extra branch for a higher version is only created when it's actually neecessary and different from the earlier one. That will also save us some purely formal shuffling of branches in repos that still work without changes.

I'll change that in the description.

Reply:
Updated. Now it suggests that merging `breaking` into `develop` will resolve the issue automatically. I have also rewritten most of it to not require adding a separate branch for `breaking` unless necessary.